backends-common/elasticsearch-v8/src/main/java/org/apache/james/backends/es/v8/IndexCreationFactory.java

```diff
@@ -0,0 +1,376 @@
+                                                                Optional<IndexSettings> indexSettings,
+                                                                Optional<TypeMapping> mappingContent) {
+            Preconditions.checkNotNull(indexName);
+            try {
+                createIndexIfNeeded(client, indexName, indexSettings.orElse(generateSetting()), mappingContent);
+                aliases.forEach(Throwing.<AliasName>consumer(alias -> createAliasIfNeeded(client, indexName, alias))
+                    .sneakyThrow());
+            } catch (IOException e) {
+                LOGGER.error("Error while creating index : ", e);
+            }
+            return client;
+        }
+
+        private void createAliasIfNeeded(ReactorElasticSearchClient client, IndexName indexName, AliasName aliasName) throws IOException {
+            if (!aliasExist(client, aliasName)) {
+                client.updateAliases(
+                    new UpdateAliasesRequest.Builder()
+                        .actions(new Action.Builder()
+                            .add(new AddAction.Builder()
+                                .index(indexName.getValue())
+                                .alias(aliasName.getValue())
+                                .build())
+                            .build())
+                        .build());
+            }
+        }
+
+        private boolean aliasExist(ReactorElasticSearchClient client, AliasName aliasName) {
+            return client.aliasExists(new ExistsAliasRequest.Builder()
+                    .name(aliasName.getValue())
+                    .build())
+                .map(BooleanResponse::value)
+                .block();
+        }
+
+        private void createIndexIfNeeded(ReactorElasticSearchClient client, IndexName indexName, IndexSettings settings, Optional<TypeMapping> mappingContent) throws IOException {
+            try {
+                if (!indexExists(client, indexName)) {
+                    CreateIndexRequest.Builder request = new CreateIndexRequest.Builder()
+                        .index(indexName.getValue())
+                        .waitForActiveShards(new WaitForActiveShards.Builder()
+                            .count(waitForActiveShards)
+                            .build())
+                        .settings(settings);
+                    mappingContent.ifPresent(request::mappings);
+                    client.createIndex(request.build())
+                        .block();
+                }
+            } catch (ElasticsearchException exception) {
+                if (exception.getMessage().contains(INDEX_ALREADY_EXISTS_EXCEPTION_MESSAGE)) {
+                    LOGGER.info("Index [{}] already exists", indexName.getValue());
+                } else {
+                    throw exception;
+                }
+            }
+        }
+
+        private boolean indexExists(ReactorElasticSearchClient client, IndexName indexName) {
```

Comment:
Should we merge `indexExists` and `createIndex` in a single pipeline?

Reply:
Let's avid too much refactorings IMO